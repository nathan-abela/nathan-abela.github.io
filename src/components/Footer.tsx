import React from 'react';

import { MdMail } from 'react-icons/md';
import { SiGithub, SiYoutube, SiLinkedin } from 'react-icons/si';
import styled from 'styled-components';

import Container from './Container';
import Grid from './Grid';
import Link from './Link';
import Text from './Text';

const StyledFooter = styled.footer`
    bottom: 0;
    left: 0;

    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;

    width: 100%;
    height: 100px;
    padding: 60px 20px 100px;
`;

const FooterGrid = styled.div`
    display: grid;
    justify-content: center;

    margin-top: 20px;
    margin-bottom: 30px;

    grid-template-columns: repeat(4, 1fr);
    gap: 30px;
`;

const Footer = (): JSX.Element => {
    const links = React.useMemo(
        () => [
            {
                url: 'https://github.com/nathan-abela',
                icon: SiGithub,
            },
            {
                url: 'https://www.youtube.com/channel/UCSmoiptI7hDaPiMq6nu4Yzg',
                icon: SiYoutube,
            },
            {
                url: 'mailto:nathanabela7@gmail.com',
                icon: MdMail,
            },
            {
                url: 'https://linkedin.com/in/nathan-abela',
                icon: SiLinkedin,
            },
        ],
        []
    );

    return (
        <StyledFooter>
            <Container paddingY='25px'>
                <Grid gridGap='30px'>
                    <Link href='/'>Home</Link>
                    <Link href='/inprogress'>About</Link>
                    <Link href='/inprogress'>Projects</Link>
                    <Link href='mailto:nathanabela7@gmail.com'>Contact</Link>
                </Grid>
            </Container>
            <FooterGrid>
                {links.map(({ url, icon: Icon }) => (
                    <Link key={url} target='_blank' opacity={0.7} href={url}>
                        <Icon size={24} />
                    </Link>
                ))}
            </FooterGrid>
            <Text margin={0} fontSize='0.9rem' color='rgba(0, 0, 0, 0.7)'>
                Â© {new Date().getFullYear()} Nathan Abela
            </Text>
        </StyledFooter>
    );
};

export default Footer;
