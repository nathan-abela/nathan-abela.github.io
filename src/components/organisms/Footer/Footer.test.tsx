import { render } from '@testing-library/react';

import { Footer } from './Footer';

describe('Footer', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<Footer />
		);
		expect(container).toMatchSnapshot();
	});
});
