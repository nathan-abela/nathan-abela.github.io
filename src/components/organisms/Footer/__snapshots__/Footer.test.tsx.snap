// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <footer
    class="styles__StyledFooter-sc-1pzkd2j-0 bbUQND"
  >
    <div
      class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
    >
      <div
        class="styles__Flex-sc-1pzkd2j-1 dfBZPt"
      >
        <span>
          Â© 
          2025
           Nathan Abela
        </span>
        <ul
          class="styles__StyledSocialIcons-sc-1vfff3-0 fBtYTT"
        >
          <li>
            <a
              href="https://github.com/nathan-abela"
              rel="noopener noreferrer"
              target="_blank"
              title="Github"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/nathan-abela"
              rel="noopener noreferrer"
              target="_blank"
              title="LinkedIn"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="https://www.youtube.com/@nathanabela"
              rel="noopener noreferrer"
              target="_blank"
              title="YouTube"
            >
              <test-file-stub />
            </a>
          </li>
          <li>
            <a
              href="mailto:nathanabela7@gmail.com"
              rel="noopener noreferrer"
              target="_blank"
              title="Email"
            >
              <test-file-stub />
            </a>
          </li>
        </ul>
      </div>
    </div>
  </footer>
</div>
`;
